manim/utils/docbuild/autocolor_directive.py

```diff
@@ -1,57 +1,98 @@
 
-    def process_doc(self, docstrings: List[List[str]]) -> Iterator[str]:
-        return []
+class ManimColorModuleDocumenter(Directive):
+    objtype = "automanimcolormodule"
+    required_arguments = 1
+    has_content = True
 
     def add_directive_header(self, sig: str) -> None:
         super().add_directive_header(sig)
 
-    def add_content(self,
-                    more_content: StringList | None,
-                    no_docstring: bool = False,
-                    ) -> None:
-
-        super().add_content(more_content, no_docstring)
-
-        source_name = self.get_sourcename()
-        self.add_line('', source_name)
-
-        self.add_line('.. list-table::', source_name)
-        self.add_line('   :header-rows: 1', source_name)
-        self.add_line('', source_name)
-        self.add_line('   * - Color name', source_name)
-        self.add_line('     - Color', source_name)
-        self.add_line('     - RGB hex code', source_name)
-
-
-        for member_name in self.get_module_members():
-            member = getattr(self.module, member_name)
-            if isinstance(member, ManimColor):
-                self.add_line(f'   * - ``{member_name}``', source_name)
-                self.add_line(f'     - .. raw:: html', source_name)
-                self.add_line(f'', source_name)
-                self.add_line(
-                    f'         <div style="background-color:{member.to_hex()};width:50px;height:2ex;"></div>',
-                    source_name
+    def run(
+        self,
+    ) -> None:
+        module_name = self.arguments[0]
+        try:
+            import importlib
+
+            module = importlib.import_module(module_name)
+        except ImportError:
+            return [
+                nodes.error(
+                    None,
+                    nodes.paragraph(text="Failed to import module '%s'" % module_name),
+                )
+            ]
+
+        # Number of Colors displayed in one row
+        num_color_cols = 2
+        table = nodes.table()
+
+        tgroup = nodes.tgroup(cols=num_color_cols * 2)
+        table += tgroup
+        for _ in range(num_color_cols * 2):
+            tgroup += nodes.colspec(colwidth=1)
+
+        # Create header rows for the table
+        thead = nodes.thead()
+        row = nodes.row()
+        for _ in range(num_color_cols):
+            col1 = nodes.paragraph(text="Color Name")
+            col2 = nodes.paragraph(text="RGB Hex Code")
+            row += nodes.entry("", col1)
+            row += nodes.entry("", col2)
+        thead += row
+        tgroup += thead
+
+        color_elements = []
+        for member_name, member_obj in inspect.getmembers(module):
+            if isinstance(member_obj, ManimColor):
+                hex_code = member_obj.to_hex()[1:]
+                r, g, b = tuple(int(hex_code[i : i + 2], 16) for i in (0, 2, 4))
+                luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255
```

Comment:
Can't Python's colorsys do this? Should HSL be supported by ManimColor too?

Reply:
Good question, if so then we should change it!

Reply:
It can't, at least not for the value you compute here. I've simplified the code a bit, regardless.